src/app/work-item/work-item-list/work-item-list.component.ts

```diff
@@ -217,28 +158,30 @@ export class WorkItemListComponent implements OnInit, AfterViewInit {
     this.broadcaster.on<string>('logout')
       .subscribe(message => {
         this.loggedIn = false;
-        this.deactiveAllFilters();
         this.authUser = null;
-        this.setFilterValues();
     });
-  }
-
-  //Detailed add functions
-  getWorkItemTypes(){
-    this.workItemService.getWorkItemTypes()
-      .then((types) => {
-        this.workItemTypes = types;
-      });
-  }
-  showTypes() {
-    this.showTypesOptions = true;
-  }
-  closePanel() {
-    this.showTypesOptions = false;
-  }
-  onChangeType(type: string) {
-    this.showTypesOptions = false;
-    this.router.navigate(['/work-item-list/detail/new?' + type]);
+    this.broadcaster.on<string>('item_filter')
+      .subscribe((filters: any) => {
+        this.filters = filters;
+        this.loadWorkItems();
+    });
+    this.broadcaster.on<string>('move_item')
```

Comment:
This way I am passing the event to work-item-list from work-item. URL is another way to pass the data. List operations should not be residing in work-item.